beacon.go

```diff
@@ -19,108 +19,95 @@ package beacon
 import (
 	"encoding/xml"
 	"errors"
-	"golang.org/x/oauth2/google"
-	"google.golang.org/api/genomics/v1"
-	"google.golang.org/appengine"
+	"fmt"
 	"net/http"
 	"strconv"
+
+	"cloud.google.com/go/bigquery"
+	"google.golang.org/appengine"
 )
 
 type beaconConfig struct {
-	variantSetIds []string
+	projectID string
+	table     string
 }
 
 var config = beaconConfig{
-	variantSetIds: []string{"3049512673186936334"},
+	projectID: "project-id",
+	table:     "genomics-public-data.platinum_genomes.variants",
 }
 
 func init() {
 	http.HandleFunc("/", handler)
 }
 
-func requestToSearch(r *http.Request) (*genomics.SearchVariantsRequest, string, error) {
-	allele := r.FormValue("allele")
-	search := &genomics.SearchVariantsRequest{
-		ReferenceName: r.FormValue("chromosome"),
-		VariantSetIds: config.variantSetIds,
+func handler(w http.ResponseWriter, r *http.Request) {
+	refName, allele, coord, err := parseInput(r)
+	if err != nil {
+		http.Error(w, "Failed parsing parameters", http.StatusBadRequest)
```

Comment:
You need to return after writing a response with http.Error

It's better to do more like the original code where you have a function that returns an error and that calls http.Error as appropriate

Reply:
done